subprojects/docs/src/docs/release/notes.md

```diff
@@ -55,6 +55,44 @@ ADD RELEASE FEATURES ABOVE
 
 -->
 
+## New features and usability improvements
+
+### Better modeling of command line arguments for compiler daemons
```

Comment:
It seems to me that we need to add something to the user manual here as well. If `ForkOptions` is part of the DSL-documentation, then `ProviderAwareForkOptions` should be as well.

Reply:
Can you be more specific as to what you think we should do here?  I couldn't find anywhere in the user manual where it seemed like it made sense to introduce this.  The Java, Groovy and Scala sections don't get into this level of detail.  Of course, we could add something about compiler daemons and touch on the argument providers - or were you thinking there was a better place for it?

And the comment about the DSL is not clear to me.  `ForkOptions` is _not_ a part of the DSL documentation as far as I can tell (i.e. they just link to the javadocs).  Or did you mean `CompileOptions` is a part of the DSL docs and so `ForkOptions` and friends should also have DSL docs?

Reply:
Here is some well hidden list of tasks which allow to add command line argument providers: https://github.com/gradle/gradle/blob/af3b4b6ffcf6ead4499c429faa75f529ce80b626/subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc#nested-inputs

We should add the new tasks/properties here as well.

Reply:
👍 I've added the new stuff to this list and created DSL pages for the different fork option objects.